Silicon/RaptorlakePkg/Library/FusaCrashLogLib/FusaCrashLogLib.c

```diff
@@ -0,0 +1,381 @@
+
+  return EFI_SUCCESS;
+}
+
+/**
+
+  Add a new Firmware CrashLog Entry to existing Error Status Block.
+
+  @param[in] ErrStsBlk                          BERT GENERIC_ERROR_STATUS instance.
+  @param[in out] FirmwareGenericErrorAddr       Pointer to this FirmwareGenericError entry address, updated on return.
+  @param[in out] FirmwareCrashLogPayloadAddr    Pointer to Firmware CrashLog Entry payload address, updated on return.
+  @param[in] EntrySize                          Firmware CrashLog Entry payload size.
+  @param[in] EntrySource                        Firmware CrashLog Entry source.
+  @param[in] EntryVersion                       Firmware CrashLog Entry version.
+
+**/
+
+VOID
+AddFirmwareCrashLogEntry (
+  IN     EFI_ACPI_6_4_GENERIC_ERROR_STATUS_STRUCTURE    *ErrStsBlk,
+  IN OUT FIRMWARE_GENERIC_ERROR                         **FirmwareGenericErrorAddr,
+  IN OUT UINT8                                          **FirmwareCrashLogPayloadAddr,
+  IN     UINT32                                         EntrySize,
+  IN     EFI_GUID                                       RecordIdGuid
+  )
+{
+  FIRMWARE_GENERIC_ERROR  *FirmwareGenericError = NULL;
+
+  FirmwareGenericError = (FIRMWARE_GENERIC_ERROR*)((UINT8*)(void*)ErrStsBlk + ErrStsBlk->DataLength + sizeof (EFI_ACPI_6_4_GENERIC_ERROR_STATUS_STRUCTURE));
+  ZeroMem (FirmwareGenericError, sizeof (FIRMWARE_GENERIC_ERROR));
+
+  CopyMem (&FirmwareGenericError->GenericErrorDataEntry.SectionType, &gEfiFirmwareErrorSectionGuid, sizeof (EFI_GUID));
+  FirmwareGenericError->GenericErrorDataEntry.ErrorSeverity = EFI_ACPI_6_4_ERROR_SEVERITY_FATAL;
+  FirmwareGenericError->GenericErrorDataEntry.Revision = EFI_ACPI_6_4_GENERIC_ERROR_DATA_ENTRY_REVISION;
+  FirmwareGenericError->GenericErrorDataEntry.ValidationBits = 0;
+  FirmwareGenericError->GenericErrorDataEntry.ErrorDataLength = EntrySize + sizeof (EFI_FIRMWARE_ERROR_DATA);
+  FirmwareGenericError->EfiFirmwareErrorData.ErrorType = 2; //FW CrashLog Error Record
+  FirmwareGenericError->EfiFirmwareErrorData.Revision = EFI_2_7_FIRMWARE_ERROR_RECORD_REVISION;
+  FirmwareGenericError->EfiFirmwareErrorData.RecordIdGuid = RecordIdGuid;
+
+  ErrStsBlk->DataLength += FirmwareGenericError->GenericErrorDataEntry.ErrorDataLength + sizeof (EFI_ACPI_6_4_GENERIC_ERROR_DATA_ENTRY_STRUCTURE);
+  ErrStsBlk->BlockStatus.ErrorDataEntryCount++;
+
+  *FirmwareGenericErrorAddr = FirmwareGenericError;
+  *FirmwareCrashLogPayloadAddr = (UINT8*)(void*) FirmwareGenericError + sizeof (FIRMWARE_GENERIC_ERROR);
+}
+
+/**
+  Get CPU Crash Log, and initialize the APEI BERT GENERIC_ERROR_STATUS structure
+
+  @param[in] ErrStsBlk            BERT GENERIC_ERROR_STATUS instance.
+
+  @retval EFI_SUCCESS             The function completes successfully
+**/
+EFI_STATUS
+GenFwBootErrorlog (
+  IN OUT EFI_ACPI_6_4_GENERIC_ERROR_STATUS_STRUCTURE    *ErrStsBlk,
+  IN OUT EFI_ACPI_6_4_BOOT_ERROR_RECORD_TABLE_HEADER    *Bert
```

Comment:
Make sure function parameters are aligned with function comments

Reply:
Parameters and function comments updated.